lib/routes/recovery-keys.js

```diff
@@ -78,10 +79,111 @@ module.exports = (log, db, Password, verifierVersion, customs) => {
           .then(() => { return {recoveryData} })
 
         function getRecoveryKey() {
-          return db.getRecoveryKey(uid, recoveryKeyId)
-            .then((res) => recoveryData = res.recoveryData)
+          return db.getRecoveryKey(uid)
+            .then((res) => {
+              // `db.getRecoveryKey` doesn't require recoveryKeyId to retrieve
+              // the recovery bundle, however, we should perform a sanity
+              // check to ensure that the returned bundle contains the recoveryKeyId.
```

Comment:
👍  this is exactly what I came to this PR to check for :-)

I would suggest not calling it a "sanity-check", the fact that the user has to provide the correct recoveryKeyId is a security feature so it's probably worth saying that explicitly in the comment.

Also, since this is checking whether the user provided a secret value, it should use a constant-time string comparison like `crypto.butil.buffersAreEqual` rather than `!==` to avoid leaking information that might help an attacker guess the correct keyid.